service/collector.go

```diff
@@ -106,7 +106,7 @@ func New(set CollectorSettings) (*Collector, error) {
 }
 
 // GetStateChannel returns state channel of the collector server.
-func (col *Collector) GetStateChannel() chan State {
+func (col *Collector) GetStateChannel() <-chan State {
```

Comment:
Can we make this just "GetState()" and don't expose any channel? Then we can have a simple atomic variable for the state.

Reply:
Im ok to do that :)

Reply:
done